workshops/quotes_generator/README.md

```diff
@@ -0,0 +1,351 @@
+
+Now let's write a function which will make the API request and fetch the data from that URL. We'll use `async functions` because it will return us a promise. We'll also use `await` inside the `async function`.
+
+![Image of async await which looks cool](https://cloud-b58j7jpbu.vercel.app/0image.png)
+
+The `await` operator is used to wait for a `Promise` to either resolve or reject. It can only be used inside an `async function`.
+
+`Async functions` can contain zero or more `await` expressions. `Await` expressions suspend progress through an `async function`, yielding control and subsequently resuming progress only when an awaited promise-based asynchronous operation is either fulfilled or rejected. The resolved value of the promise is treated as the return value of the await expression. Use of `async` / `await` enables the use of ordinary `try` / `catch` blocks around asynchronous code.
+
+Learn more about [`Async await`](https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Asynchronous/Async_await).
+
+Inside our `App` component, we'll create an `async` function `getQuotes()` and now let's start fetching the data.
+
+```jsx
+async function getQuotes() {
+  try {
+    const res = await fetch(url);
+    const data = await res.json();
+    setQuotes(data);
+  } catch (err) {
+    console.log(err);
+  }
+}
+```
+
+Explanation: Inside our `async` function, we add a `try` / `catch` block, which will help us to get the data or throw errors if not successful. We create a variable called `res` (shortform for `response`) which will store the response we get after we fetch the data from the URL. Now this response is not yet ready to use and we need to convert it to `json()` format before using it. So we store that json formatted data in the `data` variable. And lastly, we store all the data inside our `quotes` state using `setQuotes`.
+
+The `catch` block will simply catch an error (if any) and then log it to the console.
+
+![try catch example](https://cloud-oxjmrq4w2.vercel.app/0image.png)
+
+Our `async` function is complete and we can now use the data we got from the API request.
+
+Now if we take a look at the value stored inside our state, we'll notice that it is a whole big array consisting of 100s of objects containing quotes! 
+
+![Image of the state containing 1000s of quotes](https://cloud-7drh3ecj9.vercel.app/0image.png)
+
+**NOTE:** You can check this in the codeSandbox's `React DevTools` (next to the console).
+
+We only need to display 1 random quote at a time. So, for this, we only need to store a single random object from that array inside our state.
+
+Using the formula for getting a random number, let's create a function outside our component which will return us a random number.
+
+```jsx
+const randomNo = () => Math.floor(Math.random() * data.length) + 1;
```

Comment:
Can you give a little bit more context for where this code should go? Could just be in the form of something like

```js
export default function App() {
 // code from earlier workshop
}

const randomNo = () => Math.floor(Math.random() * data.length) + 1;
```